docs/sources/configure-client/grafana-agent/go_pull.md

```diff
@@ -1,258 +1,224 @@
-
-The reason for using this special package is because godeltaprof is a memory profiler specialized for collecting cumulative profiles (heap, block, mutex) efficiently. It is more efficient because it does the delta/merging before producing pprof data, avoiding extra decompression/parsing/allocations/compression.
+1. Select all pods 
+  ```river
+  discovery.kubernetes "all_pods" {
+          role = "pod"
+  }
+  ```
+2. Drop not running pods, create `namespace`, `pod`, `node` and `container` labels. 
+  Compose `service_name` label based on `namespace` and `container` labels.
+  Select only services matching regex pattern `(ns1/.*)|(ns2/container-.*0)`.
+    ```river
+    
+    discovery.relabel "specific_pods" {
+            targets = discovery.kubernetes.all_pods.targets
+    
+            rule {
+                    action        = "drop"
+                    regex         = "Succeeded|Failed|Completed"
+                    source_labels = ["__meta_kubernetes_pod_phase"]
+            }
+    
+            rule {
+                    action        = "replace"
+                    source_labels = ["__meta_kubernetes_namespace"]
+                    target_label  = "namespace"
+            }
+    
+            rule {
+                    action        = "replace"
+                    source_labels = ["__meta_kubernetes_pod_name"]
+                    target_label  = "pod"
+            }
+    
+            rule {
+                    action        = "replace"
+                    source_labels = ["__meta_kubernetes_node_name"]
+                    target_label  = "node"
+            }
+    
+            rule {
+                    action        = "replace"
+                    source_labels = ["__meta_kubernetes_pod_container_name"]
+                    target_label  = "container"
+            }
+    
+            rule {
+                    action        = "replace"
+                    regex         = "(.*)@(.*)"
+                    replacement   = "${1}/${2}"
+                    separator     = "@"
+                    source_labels = ["__meta_kubernetes_namespace", "__meta_kubernetes_pod_container_name"]
+                    target_label  = "service_name"
+            }
+            
+            rule { 
+                    action        = "keep"
+                    regex         = "(ns1/.*)|(ns2/container-.*0)"
+                    source_labels = ["service_name"]
+            }
+    }
+    ```
 
-To start using godeltaprof in pull mode in a Go application, you need to include godeltaprof module in your app:
+3. And then use `discovery.relabel.specific_pods.targets` as a target for `pyroscope.scrape` block.
 
-```bash
-go get github.com/grafana/pyroscope-go/godeltaprof@latest
-```
+    ```river
+        pyroscope.scrape "scrape_job_name" {
+                targets    = discovery.relabel.specific_pods.output
+                ...
+        }
+    ```
 
-and add it to your imports:
+### Exposing pprof endpoints
```

Comment:
Should this be at the end or the beginning? seems like we should have people expose their pprof endpoints and _then_ do the rest of the steps to scrape from them

Reply:
either way, I think we should put this next to, if not in, the "exposing pprof endpoints" section. I suspect someone who was looking for this would expect to find it near or in that section

Reply:
lets remove it alltogether. I dont want this to be next to the  "exposing pprof endpoints" as it complicates it